src/market/directory.rs

```diff
@@ -225,7 +255,7 @@ pub fn start_directory_server(directory: Arc<DirectoryServer>) {
     // Write the addresses to file
     let file_content = addresses
```

Comment:
Write the `self.addresses` to the file, not this new variable. Having new variable could create problem later on. Where you can have `self.addresses` updated, but the new `address` variable isn't. To remove possibility of such situation just always read and write to `self.addresses`.